### PR TITLE
8345433: (fs) Use stream to load FileTypeDetectors in Files.probeContentType

### PR DESCRIPTION
**Pull request by bplb**
Change `FileTypeDetector` list creation to use a stream rather than an iteration and adds.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8345433](https://bugs.openjdk.org/browse/JDK-8345433): (fs) Use stream to load FileTypeDetectors in Files.probeContentType (**Enhancement** - P4)

### Reviewers
 * [Roger Riggs](https://openjdk.org/census#rriggs) (@RogerRiggs - **Reviewer**)
 * [Alan Bateman](https://openjdk.org/census#alanb) (@AlanBateman - **Reviewer**)
 * [Andrey Turbanov](https://openjdk.org/census#aturbanov) (@turbanoff - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22552/head:pull/22552` \
`$ git checkout pull/22552`

Update a local copy of the PR: \
`$ git checkout pull/22552` \
`$ git pull https://git.openjdk.org/jdk.git pull/22552/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22552`

View PR using the GUI difftool: \
`$ git pr show -t 22552`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22552.diff">https://git.openjdk.org/jdk/pull/22552.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22552#issuecomment-2517981401)
</details>

### CODE REVIEW

**Comment by bplb** [comment]
This change was suggested [here](https://github.com/openjdk/jdk/pull/22132#discussion_r1844185396) by @RogerRiggs.

An unused variable in `AsynchronousChannelProvider` is incidentally removed as suggested [here](https://github.com/openjdk/jdk/pull/22132#discussion_r1847879204) by @turbanoff.

File: src/java.base/share/classes/java/nio/file/Files.java [hunk 1]
```diff
@@ -1522,13 +1522,10 @@ private static FileTypeDetector createDefaultFileTypeDetector() {
...
-            ServiceLoader<FileTypeDetector> loader = ServiceLoader
-                .load(FileTypeDetector.class, ClassLoader.getSystemClassLoader());
-            for (FileTypeDetector detector: loader) {
-                list.add(detector);
-            }
-            return list;
+            return ServiceLoader.load(FileTypeDetector.class,
+                                      ClassLoader.getSystemClassLoader())
+                .stream()
+                .map(p -> p.get()).toList();
```
Comment (turbanoff) [inline src/java.base/share/classes/java/nio/file/Files.java:1528]: Let's move `.toList();` to a separate line

Comment (bplb) [reply src/java.base/share/classes/java/nio/file/Files.java]: So changed in [44401e7](https://github.com/openjdk/jdk/pull/22552/commits/44401e788d6cd9f3b67a0ba8a67874e8aca7ba6e).

File: src/java.base/share/classes/java/nio/file/Files.java [hunk 2]
```diff
@@ -1522,13 +1522,11 @@ private static FileTypeDetector createDefaultFileTypeDetector() {
...
-            ServiceLoader<FileTypeDetector> loader = ServiceLoader
-                .load(FileTypeDetector.class, ClassLoader.getSystemClassLoader());
-            for (FileTypeDetector detector: loader) {
-                list.add(detector);
-            }
-            return list;
+            return ServiceLoader.load(FileTypeDetector.class,
+                                      ClassLoader.getSystemClassLoader())
+                .stream()
+                .map(p -> p.get())
```
Comment (AlanBateman) [inline src/java.base/share/classes/java/nio/file/Files.java:1528]: This looks fine, the alternative is to use a method ref, as in map(ServiceLoader.Provider::get) if you want.

Comment (bplb) [reply src/java.base/share/classes/java/nio/file/Files.java]: So changed in c6f997e.

**Review by AlanBateman — APPROVED** [review]

**Review by RogerRiggs — APPROVED** [review]

**Comment by bplb** [comment]
Thanks @RogerRiggs and @AlanBateman for the prompt approvals.

**Review by turbanoff — APPROVED** [review]

**Pull request closed without merging**